backoffice/src/Repositories/ImageRepository.cs

```diff
@@ -0,0 +1,91 @@
+using Droits.Clients;
+using Droits.Data;
+using Droits.Exceptions;
+using Droits.Helpers;
+using Droits.Models.Entities;
+using Droits.Models.FormModels;
+using Droits.Services;
+using Microsoft.EntityFrameworkCore;
+
+namespace Droits.Repositories;
+
+public interface IImageRepository
+{
+    Task<Image> AddAsync(Image image);
+    Task<Image> UpdateAsync(Image image);
+    Task<Image> GetImageAsync(Guid id);
+    Image GetRandomImage();
+    Task UploadImageFileAsync(Guid id, IFormFile imageFile);
+    Task<Stream> GetImageStreamAsync(string key);
+    Task DeleteImagesForWreckMaterialAsync(Guid wmId, IEnumerable<Guid> imagesToKeep);
+}
+
+public class ImageRepository : BaseEntityRepository<Image>, IImageRepository
+{
+    private readonly IS3Client _client; 
+    public ImageRepository(DroitsContext dbContext, ICurrentUserService currentUserService, IS3Client client) : base(dbContext,currentUserService)
+    {
+        _client = client;
+    }
+    
+    public async Task<Image> GetImageAsync(Guid id)
+    {
+        var image = await Context.Images
+            .FirstOrDefaultAsync(i => i.Id == id);
+
+        if ( image == null )
+        {
+            throw new ImageNotFoundException();
+        }
+
+        return image;
+    }
+    
+    public Image GetRandomImage()
+    {
+        var image = Context.Images
+            .First();
+
+        if ( image == null )
+        {
+            throw new ImageNotFoundException();
+        }
+
+        return image;
+    }
+
+
+    public async Task UploadImageFileAsync(Guid id, IFormFile imageFile)
+    {
+        var key = $"{id}_{Guid.NewGuid()}.{imageFile.FileName}";
```

Comment:
I'm thinking about this, we might want to be a bit smarter with the key 

currently we have

bucket 
 - Image
 - Image
 - Image
 
 
 maybe we want 
 
 bucket 
   - WreckMaterialImages
      - Image
      - Image
      
      
      or maybe even 
      

bucket 
   - WreckMaterialImages
      - WreckMaterialId
           - Image
           - Image
     - WreckMaterialId
          - Image
          - Image
          
          
          or even...
          
          
 bucket 
    -Droits
      - DroitId
      - WreckMaterials
         - WreckMaterialId
             - Images
               - Image
               - Image
       - WreckMaterialId
            - Images
              - Image
              - Image
              
 Which would make it really easy if someone raised a FOI req for all the files they hold on them 
 
 
 so key would be something like 
 
 var key = $"Droits/{droitId}/WreckMaterials/{WreckMaterialId}/Images/{id}_{imageFile.FileName}";